### DISCUSSION TITLE
feature: ASGI compatibility

### DISCUSSION THREAD

**Discussion by laundmo** [lona-web-org/lona#220] [Ideas]
Currently, Lona uses aiohttp as a server. Aiohttp server is slower than most ASGI libraries and doesn't use a standardized interface.

What are your thoughts on eventually making this ASGI compatible, so it can run in common ASGI servers like uvicorn?

Im thinking Starlette would be a good option as a server framework, as it's used in FastAPI which is decently well established and proven to be stable and performant.

**Comment by vcokltfre** [comment]
Laundmo just introduced me to this project and it looks like something I'd definitely be interested in using, but with it using aiohttp as a backend rather than having it as an ASGI compatible app I can run with Uvicorn it's hard for me to justify using it. (This is largely due to the huge number of libraries that do work with Starlette/ASGI apps, such as starlette-discord for Discord OAuth login, for example)

**Comment by fscherf** [comment]
@laundmo, @vcokltfre sounds reasonable. Can you provide a PR that adds ASGI compatibility?

**Comment by laundmo** [comment]
Could you tell us which are the main points where the serverside code interfaces with the library itself? in other words: what classes would we have to replace to fully change out aiophttp as a server?

**Comment by fscherf** [comment]
@laundmo Sure! The only parts that interface with aiohttp directly are ``lona/server.py`` and ``lona/command_line/run_server.py``

**Comment by fscherf** [comment]
@laundmo, @vcokltfre: Do you guys have code to look at?

**Comment by laundmo** [comment]
@fscherf im sorry to say: no. i've been quite busy and didn't have any time.

**Comment by fscherf** [comment]
@laundmo don't be sorry! :) I just wanted to make sure you are not stuck on something

**Comment by fscherf** [comment]
@laundmo: this issue is open pretty long now. Can i transfer it to the ideas section?

**Comment by laundmo** [comment]
Sure, i doubt I'll be working on it anytime soon.
